spec.html

```diff
@@ -7752,6 +7752,101 @@ <h1>Jobs and Host Operations to Enqueue Jobs</h1>
       <p>Host environments are not required to treat Jobs uniformly with respect to scheduling. For example, web browsers and Node.js treat Promise-handling Jobs as a higher priority than other work; future features may add Jobs that are not treated at such a high priority.</p>
     </emu-note>
 
+    <emu-clause id="sec-jobcallback-records">
+      <h1>JobCallback Records</h1>
+      <p>The JobCallback is a Record value used to store a callable object and a host-defined value. Callable objects that are invoked via a Job enqueued by the host may have additional host-defined context. To propagate the state, Job Abstract Closures should not capture and call callable objects directly. Instead, use HostMakeJobCallback and HostCallJobCallback (or its convenience wrapper CallJobCallback).</p>
+      <emu-note>
+        <p>For example, WHATWG HTML uses the host-defined value to propagate the incumbent settings object for Promise callbacks.</p>
+      </emu-note>
+      <p>JobCallback Records have the fields listed in <emu-xref href="#table-jobcallback-records"></emu-xref>.</p>
+      <emu-table id="table-jobcallback-records" caption="JobCallback Record Fields">
+        <table>
+          <tbody>
+            <tr>
+              <th>
+                Field Name
+              </th>
+              <th>
+                Value
+              </th>
+              <th>
+                Meaning
+              </th>
+            </tr>
+            <tr>
+              <td>
+                [[Callback]]
+              </td>
+              <td>
+                A callable object
+              </td>
+              <td>
+                The callback to invoke when the Job is invoked.
+              </td>
+            </tr>
+            <tr>
+              <td>
+                [[HostDefined]]
+              </td>
+              <td>
+                Any, default value is *undefined*.
```

Comment:
Any ECMAScript value? or anything? (and if anything, why is the default not `~empty~`)

Reply:
I could go either way.

Host-defined fields are special in they are actually *any*, not any ECMAScript value, or any specification value, or any host specification value, but literally any anything. We don't have a good placeholder value for that.

`~empty~` is a placeholder for specification values, and `undefined` is a placeholder for language values. As I said I'm fine with whichever one we use as a default.

Reply:
If it's never exposed to runtime JS, i'd lightly prefer `~empty~` as the default value.

Reply:
wfm